library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -97,6 +97,52 @@ def initialize(service)
       @errors = {}
     end
 
+    # Get the current start_mode
+    def current_start_mode
+      @current_start_mode ||=
+        if service.enabled?
+          :on_boot
+        elsif socket && socket.enabled?
+          :on_demand
+        else
+          :manual
+        end
+    end
+
+    # TODO
+    def current_active?
```

Comment:
well, I have one problem with it. How to recognize here if restart or reload is selected? this allows only to distinguish between start and stop.

Reply:
Ah, yes. I forgot the #action method.